quic/s2n-quic-core/src/inet/checksum.rs

```diff
@@ -0,0 +1,318 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::{fmt, hash::Hasher, num::Wrapping};
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+mod x86;
+
+/// Computes the [IP checksum](https://www.rfc-editor.org/rfc/rfc1071) over the given slice of bytes
+#[inline]
+pub fn checksum(data: &[u8]) -> u16 {
+    let mut checksum = Checksum::default();
+    checksum.write(data);
+    checksum.finish()
+}
+
+/// Minimum size for a payload to be considered for platform-specific code
+const LARGE_WRITE_LEN: usize = 32;
```

Comment:
I think we'd always be above this with the `min_indistinguishable_packet_len` being enforced. Though I'm fine keeping this here as well

Reply:
So that's for the whole payload, yes. But smaller values (like just the IPv4 pseudo headers) can also be written.